src/XR/features/WebXRControllerPointerSelection.ts

```diff
@@ -366,7 +370,29 @@ export class WebXRControllerPointerSelection extends WebXRAbstractFeature {
                     scene.pointerY = this._screenCoordinatesRef.y;
                 }
             }
-            controllerData.pick = this._scene.pickWithRay(controllerData.tmpRay, this._scene.pointerMovePredicate || this.raySelectionPredicate);
+
+            let utilityScenePick = null;
+            if (this._scene._utilityLayerRenderer) {
+                utilityScenePick = this._scene._utilityLayerRenderer.utilityLayerScene.pickWithRay(
+                    controllerData.tmpRay,
+                    this._scene._utilityLayerRenderer.utilityLayerScene.pointerMovePredicate || this.raySelectionPredicate
+                );
+            }
+
+            let originalScenePick = this._scene.pickWithRay(controllerData.tmpRay, this._scene.pointerMovePredicate || this.raySelectionPredicate);
+            if (!utilityScenePick || utilityScenePick.distance === 0) {

```

Comment:
Why not check for pick.hit ?

Reply:
I took code from the utility layer pick listener. But you're right pick.hit is clearer